### PR TITLE
readme: initial overhaul

### PR DESCRIPTION
**Pull request by Sirupsen**
I'll be adding a section on how it works and how to pick good configuration later. For now I've just added standard Usage documentation.

@csfrancis @byroot

### CODE REVIEW

**Comment by byroot** [comment]
Nice!

File: README.md [hunk 1]
```diff
@@ -1,83 +1,107 @@
...
+2. `#semian_identifier`. This is responsible for returning a symbol that
+   represents every unique resource, for example `redis_master` or
+   `mysql_shard_1`. This is usually assembled from a `name` attribute on the
+   Semian configuration hash, but could also be `<host>:<port>`.
+3. `connect`. The name of this method varies. You must override the driver's
+   connect method with one that wraps the connect call with
+   `Semian::Resource#acquire`. You should do this at the lowest possible level.
+4. `query`. Same as `connect` but for queries on the resource.
+
+The best resource is looking at the [already implemented adapters](#adapters).
```
Comment (byroot) [inline README.md]: You also must define the resource exceptions: `ResourceBusyError` and `CircuitOpenError`. And it's better if they inherits from the adapter base exception.

Comment (Sirupsen) [inline README.md]: Great point. Explaining that pattern is helpful!

**Comment by Sirupsen** [comment]
I'm keeping track of which documentation to update here https://github.com/Shopify/semian/issues/31

Elaborating on our patterns of the exceptions would be helpful too. But that's another PR

**Comment by csfrancis** [comment]
:100: great.

**Pull request merged**